File: 11849 CD/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        int N;
		StringBuilder output = new StringBuilder();
		while(st.nextToken() == StreamTokenizer.TT_NUMBER && (N = (int)st.nval) != 0) {
		    st.nextToken();
		    int M = (int)st.nval;
		    int[] Jack = new int[N], Jill = new int[M];
		    for(int i = 0; i < N; ++i) {
		        st.nextToken();
		        Jack[i] = (int)st.nval;
		    }
		    for(int i = 0; i < M; ++i) {
		        st.nextToken();
		        Jill[i] = (int)st.nval;
		    }
		    int i = 0, j = 0, count = 0;
		    while(i < N && j < M) {
                if(Jack[i] == Jill[j]) {
                    ++count;
                    ++i;
                    ++j;
                } else if(Jack[i] > Jill[j])
                    ++j;
                else
                    ++i;
            }
            output.append(count).append('\n');
		}
		System.out.print(output);
	}
}

File: 11849 CD/solution.c
#include <stdio.h>

int main() {
    int N, M;
    while(scanf("%d %d", &N, &M) && N != 0) {
        int Jack[N], Jill[M];
        for(int i = 0; i < N; ++i)
            scanf("%d", &Jack[i]);
        for(int i = 0; i < M; ++i)
            scanf("%d", &Jill[i]);
        int i = 0, j = 0, count = 0;
        while(i < N && j < M) {
            if(Jack[i] == Jill[j]) {
                ++count;
                ++i;
                ++j;
            } else if(Jack[i] > Jill[j])
                ++j;
            else
                ++i;
        }
        printf("%d\n", count);
    }
	return 0;
}
